packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -0,0 +1,181 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { FieldKey, EmptyKey } from "../../tree";
+import { IEditableForest, TreeNavigationResult, ITreeSubscriptionCursor } from "../../forest";
+import { TreeSchemaIdentifier } from "../../schema-stored";
+
+export const proxySymbol = Symbol.for("editable-tree");
+
+/**
+ * A dummy placeholder for a Proxy target.
+ */
+export interface IEditableTree {}
+
+/**
+ * ObjectEditableTree is a Proxy target.
+ * It holds an instance of an allocated Cursor and its child nodes to traverse the Forest.
+ */
+class ObjectEditableTree implements IEditableTree {
+	public readonly cursor: ITreeSubscriptionCursor;
+	private readonly _children: Map<string, IEditableTree>;
+
+	constructor(
+		public readonly forest: IEditableForest,
+		_cursor?: ITreeSubscriptionCursor,
+	) {
+		if (_cursor) {
+			this.cursor = _cursor.fork();
+		} else {
+			this.cursor = forest.allocateCursor();
+			forest.tryMoveCursorTo(forest.root(forest.rootField), this.cursor);
+		}
+		this._children = new Map();
+	}
+
+	public get type(): TreeSchemaIdentifier {
+		return this.cursor.type;
+	}
+
+	public getChildNode(key: FieldKey): IEditableTree | undefined {
+		// It was not implemented to improve performance or for any other "user-related" purpose.
+		// Cache might be needed to follow-up all the cursors created.
+		// Still an open question if it is a good idea.
+		if (!this._children.has(key as string)) {
+			const proxy = proxify(this.forest, this.cursor);
+			this._children.set(key as string, proxy);
+		}
+		return this._children.get(key as string);
+	}
+}
+
+const tryMoveDown = (cursor: ITreeSubscriptionCursor, key: FieldKey):
+	{ result: TreeNavigationResult; isArray: boolean; } => {
+	let result = cursor.down(key, 0);
+	let isArray = !!cursor.length(EmptyKey);
+	// TODO make better
+	if (result === TreeNavigationResult.NotFound) {
+		// reading an array
+		result = cursor.down(EmptyKey, Number(key));
+	} else {
+		if (isArray && cursor.down(EmptyKey, 0) === TreeNavigationResult.Ok) {
+			cursor.up();
+		} else {
+			isArray = false;
+		}
+	}
+	return { result, isArray };
+};
+
+function accessArrayCursor(cursor: ITreeSubscriptionCursor) {
+	return function* () {
+		const length = cursor.length(EmptyKey);
+		let nextIndex = 0;
+		while (nextIndex < length) {
+			cursor.down(EmptyKey, nextIndex++);
+			const value = cursor.value;
+			cursor.up();
+			yield value;
+		}
```

Comment:
This is much less efficient that using cursor.seek(1). See mapCursorField for an example.

Also consider not doing this with lazy iteration. If anyone modifies the cursor while this is iterating it will do really strange things. If you keep this pattern, you may want to document that limitation.

Might be better to just replace the whole thing with `return mapCursorField(cursor, (c) => c.value)`.

If keeping this code using up and down, you need to check the returned results. If this encountered "partial" data (thats not downloaded) it would creep up the tree one level after duplicating the value.
mapCursorField handles this correctly so you might want to just use it. I'm also working on revising the cursor API to avoid this pitfall, see https://github.com/microsoft/FluidFramework/pull/11610

Reply:
Ok, thanks for the info! Will update.